client.go

```diff
@@ -72,6 +91,36 @@ func (cc *ClientConn) invoke(
 		log.Panic().Msg("Invoke opts unsupported")
 	}
 
+	var err error
+	var statsBegin *stats.Begin
+	for _, sh := range cc.statsHandlers {
+		ctx = sh.TagRPC(ctx, &stats.RPCTagInfo{
+			FullMethodName: method,
+		})
+
+		beginTime := time.Now()
+		statsBegin = &stats.Begin{
+			Client:         true,
+			BeginTime:      beginTime,
+			IsClientStream: false,
+			IsServerStream: false,
+		}
+		sh.HandleRPC(ctx, statsBegin)
+	}
+	defer func() {
+		for _, sh := range cc.statsHandlers {
+			end := &stats.End{
+				Client:    true,
+				BeginTime: statsBegin.BeginTime,
```

Comment:
Below you cache `beginTime := time.Now()` once and use that instead of `statsBegin` which makes more sense to me but that's just an opinion. 

At first I thought you could have an unitialised pointer segfault at `statsBegin.BeginTime` but that would only be if there were no stats handlers and that code wouldn't run in the first place... but I had to think about it

Reply:
yeah, I agree what you say is more sane.